lib/Sema/CSDiag.cpp

```diff
@@ -2743,11 +2743,24 @@ diagnoseUnviableLookupResults(MemberLookupResult &result, Type baseObjTy,
           return;
         }
       }
-        
+
+      // Check whether the instance member is declared on parent context and if so
+      // provide more specialized message
+      auto memberTypeContext = member->getDeclContext()->getInnermostTypeContext();
+      auto currentTypeContext = CS.DC->getInnermostTypeContext();
+      if (memberTypeContext->getSyntacticDepth() < currentTypeContext->getSyntacticDepth()) {
+        diagnose(loc, diag::could_not_use_instance_member_of_outer_type_on_nested_type,
+                 memberTypeContext->getDeclaredTypeOfContext(), memberName,
+                 currentTypeContext->getDeclaredTypeOfContext()
+         ).highlight(baseRange).highlight(nameLoc.getSourceRange());
```

Comment:
Nit: This is stdlib-style formatting.  Please move the dangling paren to the end of the previous line and indent the member call like `highlight` below.